src/common.c

```diff
@@ -1336,7 +1336,11 @@ ssize_t ofi_bsock_recv(struct ofi_bsock *bsock, void *buf, size_t len)
 		return bytes + ret;
 
 out:
-	assert(ret != -OFI_EINPROGRESS_URING);
+	if (ret == -OFI_EINPROGRESS_URING) {
+		assert(!bytes);
+		return ret;
+	}
```

Comment:
This comment is for the commit message, not this section of code.  The bsock prefetching receive data is helpful when dealing with a large number of small messages.